ts/Accessibility/Components/LegendComponent.ts

```diff
@@ -535,21 +535,32 @@ class LegendComponent extends AccessibilityComponent {
             response = keyboardNavigationHandler.response,
             chart = this.chart,
             a11yOptions = chart.options.accessibility,
+            wrapAround = a11yOptions.keyboardNavigation.wrapAround,
             numItems = chart.legend.allItems.length,
-            direction = (keyCode === keys.left || keyCode === keys.up) ? -1 : 1;
+            direction = (
+                keyCode === keys.left || keyCode === keys.up
+            ) ? -1 : 1,
+            highlightedLegendItemIx = this.highlightedLegendItemIx,
+            rawDirectionalInput = highlightedLegendItemIx + direction,
+            adjustedDirection = ((
+                rawDirectionalInput > -1 &&
+                rawDirectionalInput < numItems
+            ) ? direction : (
+                    wrapAround && (
+                        (numItems - 1) * (direction > 0 ? -1 : 1)
+                    ) || 0
+                )
+            ),
+            res = chart.highlightLegendItem(
+                highlightedLegendItemIx + adjustedDirection
+            ) || void 0;
```

Comment:
I think we don't need that `OR` with `undefined` value. After removing that everything works as expected.

Reply:
Good catch!